src/python/grpcio_health_checking/grpc/health/v1alpha/health.py

```diff
@@ -0,0 +1,73 @@
+import abc
+import enum
+from grpc.health.v1alpha import health_pb2
+
+
+@enum.unique
+class HealthStatus(enum.Enum):
+  UNKNOWN = health_pb2.HealthCheckResponse.UNKNOWN
+  SERVING = health_pb2.HealthCheckResponse.SERVING
+  NOT_SERVING = health_pb2.HealthCheckResponse.NOT_SERVING
+
+
+class _HealthServicer(health_pb2.EarlyAdopterHealthServicer):
+
+  def __init__(self):
+    self._server_status = {}
+
+  def Check(self, request, context):
+    server_key = (request.host, request.service,)
+    return health_pb2.HealthCheckResponse(
+        status=self._server_status[server_key].value)
+
+  def set(host, service, status):
+    if not isinstance(status, HealthStatus):
+      raise TypeError('expected grpc.health.v1alpha.health.HealthStatus')
```

Comment:
A good illegal-value message informs its recipient not only of what values are legal but also of what particular illegal value was the cause of the message. :-)